src/extract-args-types/extract-arg-types.ts

```diff
@@ -0,0 +1,100 @@
+interface Property {
+    name: string;
+    defaultValue?: string;
+    type: string;
+    description: any /* should be string, is currently string[] */;
+}
+
+type Inputs = Array<Property>;
+type Outputs = Array<Property>;
+type PropertiesWithoutDecorator = Array<Property>;
+
+interface ExtractedDirective {
+    inputs: Inputs;
+    outputs: Outputs;
+    propertiesWithoutDecorators: PropertiesWithoutDecorator;
+}
+
+type ExtractedDirectives = {
+    [key: string]: ExtractedDirective;
+};
+
+// probably typed somewhere else
+type ArgsTableProps = { [key: string]: ArgsTableProp };
+type ArgsTableProp = {
+    defaultValue: string | undefined;
+    description: string;
+    name: string;
+    type: {
+        name: string;
+        required: boolean;
+    };
+    table: {
+        category: "properties" | any;
+        defaultValue: { summary: string | undefined };
+        type: {
+            required: boolean;
+            summary: string | undefined;
+        };
+    };
+};
+
+const getAngularDirectiveProperties = (
+    name: string
+): ExtractedDirective | undefined =>
+    // eslint-disable-next-line no-undef
+    (window as any).STORYBOOK_ANGULAR_ARG_TYPES[name];
+
+const mapPropToArgsTableProp = (
+    prop: Property,
+    category: "inputs" | "outputs" | "properties"
+): ArgsTableProp => ({
+    name: prop.name,
+    description: prop.description,
+    defaultValue: prop.defaultValue,
+    type: { name: prop.type, required: true }, // todo get required
+    table: {
+        defaultValue: {
+            summary: prop.defaultValue,
+        },
+        type: {
+            summary: prop.type,
+            required: true, // todo get required
+        },
+        category,
+    },
+});
+
+// eslint-disable-next-line
+const mapPropsToArgsTableProps = (directive: ExtractedDirective): any => {
+    // const argsTableProps: ArgsTableProps = {};
+    const argsTableProps = [];
+
+    // eslint-disable-next-line ,no-restricted-syntax,no-restricted-syntax
```

Comment:
use outer for-loop with `for (['inputs', 'outputs', ...])`  to remove redundant code